### ISSUE TITLE
[wx] DB file is not locked when opened via menu

### ISSUE THREAD

**Issue by nobugshere** [pwsafe/pwsafe#1312]
When a file is opened via the "File->Open..." menu or the Open button, no lock file is created.

**Comment by rafaelx** [comment]
Wasn't this problem solved by #1311?

**Comment by nobugshere** [comment]
No. #1311 removes the lock on the file that is being "closed".  This issue is that the file being opened, via the File|Open menu, is not getting locked at all.  This looks to be a bit more complicated because the code to check and set the lock is already duplicated, so it really should be consolidated, perhaps merged into a new "FileLock" class.  Also, I was kind of confused by the R/O status handling.

**Comment by nobugshere** [comment]
I discovered this had already been reported in #545.
